<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-1-g15f4949  Build Version: 1.6.7594.29634
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="gantry_assembly_v2">
  <link name="world"/>
  <joint 
    name="world_to_dummy_joint"
    type="fixed">
    <parent link="world"/>
    <child link="dummy_link"/>
  </joint>

  <link name="dummy_link"/>
  <joint 
    name="dummy_joint"
    type="fixed">
    <parent link="dummy_link"/>
    <child link="gantry_base"/>
  </joint>

  <link
    name="gantry_base">
    <inertial>
      <origin
        xyz="0.761999999999998 1.0414 1.12896315789474"
        rpy="0 0 0" />
      <mass
        value="261.537541439999" />
      <inertia
        ixx="250.903905040112"
        ixy="2.77435019402361E-13"
        ixz="-1.47257562822484E-13"
        iyy="163.69656521287"
        iyz="-2.95263846178068E-13"
        izz="276.537389580734" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gantry_base.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gantry_base.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="h_slider">
    <inertial>
      <origin
        xyz="-0.184255400209413 0.0635000000000032 -3.10862446895044E-15"
        rpy="0 0 0" />
      <mass
        value="11.96255672" />
      <inertia
        ixx="2.27659914886572"
        ixy="-1.31950083718528E-18"
        ixz="-6.53826116987587E-16"
        iyy="2.26399170710001"
        iyz="-8.84639806254247E-19"
        izz="0.0246245979980174" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/h_slider.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/h_slider.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="h_slider_joint"
    type="prismatic">
    <origin
      xyz="0.761999999999999 1.651 1.4986"
      rpy="1.5707963267949 0 1.5707963267949" />
    <parent
      link="gantry_base" />
    <child
      link="h_slider" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.04"
      upper="0.22"
      effort="1"
      velocity="1" />
  </joint>
  <link
    name="v_slider">
    <inertial>
      <origin
        xyz="-2.22044604925031E-16 0.0412906505115238 -0.247755400209413"
        rpy="0 0 0" />
      <mass
        value="7.49764372086085" />
      <!-- <mass
        value="0.49764372086085" /> -->
      <inertia
        ixx="0.81898723375128"
        ixy="3.10681166828544E-17"
        ixz="9.58825360543054E-19"
        iyy="0.014274274403217"
        iyz="-8.75329203593836E-18"
        izz="0.830051122898296" />

      <!-- <inertia
        ixx="0.000081898723375128"
        ixy="0.00010681166828544E-17"
        ixz="9.58825360543054E-19"
        iyy="0.00014274274403217"
        iyz="-8.75329203593836E-18"
        izz="0.000830051122898296" /> -->
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/v_slider.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/v_slider.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="v_slider_joint"
    type="prismatic">
    <origin
      xyz="0.0635000000000001 0.501641895675784 0"
      rpy="0 1.5707963267949 0" />
    <parent
      link="h_slider" />
    <child
      link="v_slider" />
    <axis
      xyz="0 1 0" />
   
    <limit
      lower="-0.80"
      upper="-0.02"
      effort="0"
      velocity="1" />
    <dynamics damping="1" friction="1"/>
  </joint>
  <link
    name="gripper_base">
    <inertial>
      <origin
        xyz="-0.00338422894179746 -0.222329644889651 -0.527028229762171"
        rpy="0 0 0" />
      <mass
        value="24.6210017191392" />
      <inertia
        ixx="0.177788074167716"
        ixy="-1.12340692304258E-14"
        ixz="-1.71877111598032E-17"
        iyy="5.28982422325627"
        iyz="1.09117547322104E-17"
        izz="5.44377315799644" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_base.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_base.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="v_slider_gripper_base_joint"
    type="fixed">
    <origin
      xyz="0 -1.0922 -0.0254"
      rpy="1.5708 -0.01522 0" />
    <parent
      link="v_slider" />
    <child
      link="gripper_base" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="gripper_1">
    <inertial>
      <origin
        xyz="-0.0180961792278745 0.632079909330802 0.260429644889651"
        rpy="0 0 0" />
      <mass
        value="0.777361348500002" />
      <inertia
        ixx="0.0128214004523577"
        ixy="0.000208888774010348"
        ixz="-2.16840434497101E-19"
        iyy="0.000422712497739523"
        iyz="-1.91429446079472E-19"
        izz="0.012485118359914" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_1_joint"
    type="prismatic">
    <origin
      xyz="0.165016422257285 0.0381000000000027 0.253999999999988"
      rpy="-1.5707963267949 0 3.14159265358979" />
    <parent
      link="gripper_base" />
    <child
      link="gripper_1" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-0.55601"
      upper="0"
      effort="1"
      velocity="1" />
  </joint>
  <link
    name="gripper_2">
    <inertial>
      <origin
        xyz="-0.0180961792294174 0.632079909330803 -0.184229644889656"
        rpy="0 0 0" />
      <mass
        value="0.777361348500003" />
      <inertia
        ixx="0.0128214004523577"
        ixy="0.000208888774010314"
        ixz="4.87890977618477E-19"
        iyy="0.000422712497739523"
        iyz="1.03648832576259E-18"
        izz="0.012485118359914" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://gantry_assembly_v2/meshes/gripper_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_2_joint"
    type="prismatic">
    <origin
      xyz="-0.17797357162225 -0.0381000000000005 0.253999999999976"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="gripper_base" />
    <child
      link="gripper_2" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-0.54305"
      upper="0.012957"
      effort="1"
      velocity="1" />
  </joint>

  <transmission name="trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="h_slider_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="h_prismatic">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <transmission name="trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="v_slider_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface
      </hardwareInterface>
    </joint>
    <actuator name="v_prismatic">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <transmission name="trans_3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_1_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint1_prismatic">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <transmission name="trans_4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_2_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint2_prismatic">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
    </plugin>
  </gazebo>

  <gazebo reference="v_slider">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>

  <gazebo reference="gripper_1">
    <mu1>1000</mu1>
    <mu2>1000</mu2>
    <kp>1000000.0</kp>
    <kd>100000.0</kd>
    <slip1>0</slip1>
    <slip2>0</slip2>
    <minDepth>0.001</minDepth>
    <maxVel>0</maxVel>
    <torsional>
        <ode>
            <coefficient>1000</coefficient>
            <slip>0</slip>
        </ode>
    </torsional>

</gazebo>

<gazebo reference="gripper_2">
    <mu1>1000</mu1>
    <mu2>1000</mu2>
    <kp>1000000.0</kp>
    <kd>100000.0</kd>
    <slip1>0</slip1>
    <slip2>0</slip2>
    <minDepth>0.001</minDepth>
    <maxVel>0</maxVel>
    <torsional>
        <ode>
            <coefficient>1000</coefficient>
            <slip>0</slip>
        </ode>
    </torsional>

</gazebo>
  

<gazebo>
   <plugin name="gazebo_grasp_fix" filename="libgazebo_grasp_fix.so">
        <arm>
           <arm_name> v_slider </arm_name>
           <palm_link> gripper_base  </palm_link>
           <gripper_link> gripper_1 </gripper_link>
           <gripper_link> gripper_2 </gripper_link>
           <!-- <gripper_link> ... </gripper_link> -->
        </arm>
       <forces_angle_tolerance>100</forces_angle_tolerance>
       <update_rate>4</update_rate>
       <grip_count_threshold>4</grip_count_threshold>
       <max_grip_count>8</max_grip_count>
       <release_tolerance>0.005</release_tolerance>
       <disable_collisions_on_attach>false</disable_collisions_on_attach>
       <contact_topic>__default_topic__</contact_topic>
    </plugin>
</gazebo>


</robot>